### DISCUSSION TITLE
Need to get max amps used for main breaker.

### DISCUSSION THREAD

**Discussion by Kelly12612** [jertel/vuegraf#166] [Q&A] [answered]
Thinking about geting a whole house generator, I have over a year of data, so it would be useful if I could do a query that would return the max amps (OK, watts is fine, since that's what is in the DB), ever used, or perhaps max and average by week/month, something like that.  Maybe even max and average kwh per hour, per month.  Just trying to get some idea of peak load in a way that I can map to generator capacity.
Is this a simple query, and any chance you can point me in the right direction?
Thanks!

**Comment by jertel** [comment, marked as answer]
Visualize peak kW usage of "Main Panel" over time, using per-minute data:
```
from(bucket: "vuegraf")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "energy_usage")
  |> filter(fn: (r) => r["device_name"] == "Main Panel")
  |> filter(fn: (r) => (r["detailed"] == "False"))
  |> map(fn: (r) => ({ r with _value: r._value / 1000.0}))
```

For Hourly data, change "False" to "Hour", and for Day data, change "False" to "Day"

If you want the overall single number, and don't care about the over-time visualization, add a new line with:

`|> mean()` to get the average usage overall
or
`|> max()` to get the maximum usage overall

Make sure your time window is expanded as far back as you are interested in looking.

**Reply by Kelly12612** [reply]
Thanks! FYI, I had to change the device name, mine has a string in front of it, like "XXXX Main Panel".
